training_controller/prepare_training_logs.py

```diff
@@ -176,6 +176,57 @@ def fetch_training_logs_from_elasticsearch(
         else:
             return False
 
+    def fetch_training_logs_from_opensearch(
+        self, es_instance, num_logs_to_fetch, timestamps_list
+    ):
+        # This function will customize the Elasticdump query for each interval within timestamps_list and then fetch the logs by calling the run_esdump method.
+
+        timestamps_esdump_num_logs_fetched = self.get_log_count(
+            es_instance, timestamps_list, num_logs_to_fetch
+        )
+        # ESDump logs
+        esdump_sample_command = [
+            "/usr/local/bin/elasticdump",
+            "--searchBody",
+            '{{"query": {{"bool": {{"must": [{{"match":{{"drain_error_keyword":false}}}}, {{"term": {{"is_control_plane_log": false}}}},{{"range": {{"timestamp": {{"gte": {},"lt": {}}}}}}}]}}}} ,"_source": ["masked_log", "timestamp", "is_control_plane_log", "window_start_time_ns", "_id"], "sort": [{{"timestamp": {{"order": "desc"}}}}]}}',
+            "--retryAttempts",
+            "100",
+            "--fileSize=50mb",
+            "--size={}",
+            "--limit",
+            "10000",
+            f"--input={FORMATTED_ES_ENDPOINT}/logs",
+            "--output={}",
+            "--type=data",
+        ]
+        query_queue = []
+        for idx, entry in enumerate(timestamps_list):
+            if timestamps_esdump_num_logs_fetched[idx] == 0:
+                continue
+            start_ts, end_ts, filename = (
+                entry["start_ts"],
+                entry["end_ts"],
+                entry["filename"],
+            )
+            current_command = esdump_sample_command[:]
+            current_command[2] = current_command[2].format(start_ts, end_ts)
+            current_command[6] = current_command[6].format(
+                timestamps_esdump_num_logs_fetched[idx]
+            )
+            current_command[10] = current_command[10].format(
+                os.path.join(self.ES_DUMP_DIR, f"{filename}")
+            )
+            query_queue.append(current_command)
+        """
```

Comment:
doc strings should be placed at the first line of the functions.

Reply:
That's not a doc string as much a multi-line comment discussing the last edge case so it's okay to put it here to give it context.

Reply:
if it's not a doc string , you should use `#`s to comment them..